src/page/personal-information.page.ts

```diff
@@ -0,0 +1,97 @@
+import { ElementFinder, ElementArrayFinder, element, by, browser, ExpectedConditions } from 'protractor';
+
+export class PersonalInformationPage {
+  private firstNameInput: ElementFinder;
+  private lastNameInput: ElementFinder;
+  private sexOptions: ElementArrayFinder;
+  private yearsOfExperienceOptions: ElementArrayFinder;
+  private dateInput: ElementFinder;
+  private professionOptions: ElementArrayFinder;
+  private seleniumTools: ElementArrayFinder;
+  private continents: ElementFinder;
+  private seleniumCommands: ElementFinder;
+  private button: ElementFinder;
+
+  private selectInput(labelName: string): ElementFinder {
+    return element.all(by.tagName('tr')).filter(async (elem) => {
+      const labelValue: string = await elem.element(by.tagName('span')).getText();
+      return labelValue.startsWith(labelName);
+    }).first().element(by.tagName('input'));
+  }
+
+  constructor() {
+    this.firstNameInput = element(by.name('firstname'));
+    this.lastNameInput = element(by.name('lastname'));
+    this.sexOptions = element.all(by.name('sex'));
+    this.yearsOfExperienceOptions = element.all(by.name('exp'));
+    this.dateInput = this.selectInput('Date:');
+    this.professionOptions = element.all(by.name('profession'));
+    this.seleniumTools = element.all(by.name('tool'));
+    this.continents = element(by.name('continents'));
+    this.seleniumCommands = element(by.name('selenium_commands'));
+    this.button = element(by.name('submit'));
+  }
+
+  private async selectRadioButton (
+    desiredOption: string, options: ElementArrayFinder): Promise<void> {
+    await options.filter(async (elem) => {
+      const value = await elem.getAttribute('value');
+      return value === desiredOption;
+    }).first().click();
+  }
+
+  private async selectAllCheckBoxs (
+    desiredOptions: string[], options: ElementArrayFinder): Promise<void> {
+    const optionsToSelect: ElementFinder[] = await options.filter(async (elem) => {
+      const value = await elem.getAttribute('value');
+      return desiredOptions.includes(value);
+    });
+    optionsToSelect.forEach(async (elem) => {
+      await elem.click();
+    });
+  }
+
+  private async selectContinent(desiredContinent: string): Promise<void> {
+    const selectedContinent = await this.continents.all(by.tagName('option')).filter(
+      async (elem) => {
+      const  continent = await elem.getText();
+      return continent === desiredContinent;
+    }).first();
```

Comment:
I tried with a lot of ways to find this element using this.continents.element(by.linkText(desiredContinent)) but It found it. Do you know the right way of using that selector?

Reply:
that's weird, what if you try use `this.continents.element(by.cssContainingText('option', desiredContinent))`

Reply:
That worked. Thanks!